WordPress/gutenberg
# ESLint: Fix React Compiler violations in various commands

## What?
This PR fixes most of the React Compiler ESLint violations, reducing them from 24 to 8.

## Why?
We'd like to start using the React Compiler (the babel plugin) in Gutenberg, and to get there, we'd like to enable the ESLint plugin first.

## How?
This is a syntax change, introducing a wrapper function that's tricking React Compiler that rules of hooks are being correctly followed. The underlying behavior is unchanged. We're essentially changing:

```
hook: something
```

to:

```
hook: getSomething()
```

where `getSomething()` returns `something` as well.

Ideally, we would change the API to no longer need to receive and pass a hook, but this is work for another time.

This might look like a big PR, but it's quite mundane. I recommend reviewing with whitespace changes hidden as most of the changes are actually whitespace changes.

## Testing Instructions
Verify all pre-existing commands still work by pressing cmd+k in certain contexts:
* While a block is selected: Duplicate, Add before, Add after, Group, Delete
* Any context: Add new page / Add new block
* Site editor: Navigation, Patterns, Templates, Template Parts
* Site Editor: Open styles, Learn about styles
* While list view is open: Close list view. Whie list view is closed: Open list view.
* Any editor: Enter spotlight mode, Editor preferences


### Testing Instructions for Keyboard
Same

## Screenshots or screencast
